types/stripe/index.d.ts

```diff
@@ -2153,15 +2190,31 @@ declare namespace Stripe {
                  */
                 locale?: 'auto' | 'da' | 'de' | 'en' | 'es' | 'fi' | 'fr' | 'it' | 'ja' | 'nb' | 'nl' | 'pl' | 'pt' | 'sv' | 'zh';
 
+                /**
+                 * The mode of the Checkout Session, one of payment, setup, or subscription.
+                 */
+                mode?: 'payment' | 'setup' | 'subscription';
+
                 /**
                  * Details for creation of payment intent
                  */
-                payment_intent_data?: paymentIntents.IPaymentIntentCaptureOptions | paymentIntents.IPaymentIntentData;
+                payment_intent_data?: paymentIntents.IPaymentIntentSessionSubset;
+
+                /**
+                 * A subset of parameters to be passed to SetupIntent creation.
+                 */
+                setup_intent_data?: setupIntents.ISetupIntentSessionSubset;
+                /**
+                 * Describes the type of transaction being performed by Checkout in order to customize relevant text on the page, such as the submit button.
+                 * submit_type can only be specified on Checkout Sessions using line items or a SKU, but not Checkout Sessions for subscriptions.
+                 * Supported values are auto, book, donate, or pay.
+                 */
+                submit_type?: 'auto' | 'book' | 'donate' | 'pay';
 
                 /**
                  * Use instead of @param line_items when using a subscription
                  */
-                subscription_data?: subscriptions.ISubscriptionCustCreationOptions;
+                subscription_data?: ICheckOutCreationSubscriptionData;
```

Comment:
Is the previous code wrong or only applicable to older protocol versions?  If it's still valid for some people I think the two versions should be `|`ed together.

Same question for `payment_intent_data`.

Reply:
For payment_intent_data the `paymentIntents.IPaymentIntentCaptureOptions` was my first attempt .. `paymentIntents.IPaymentIntentData` was what was there to begin with, it is clearly wrong but you are right: changing it might break code for people. I will add it again

Reply:
And on the same node, I am leaving in the original for subscription_data as well